lib/resty/websocket/client.lua

```diff
@@ -161,6 +168,15 @@ function _M.connect(self, uri, opts)
         return 1
     end
 
+    local custom_headers
+    if headers then
+      local tmp = {}
+      for k, v in pairs(headers) do
```

Comment:
Alas. `pairs()` cannot be JIT compiled and is still a NYI. Maybe we can just use a flat array like below?

```
{ key1, value1, key2, value2, ... }
```

This way we can use `ipairs()` to iterate arrays.

Reply:
That'd be a very weird format IMHO. What about `{"key: value", "key: value", ...}` -- such that we are one `concat` away from the final format?

Reply:
@salazar Fine.